File: src/apps/jhonny_home_studio_app/lib/core/utils/media_url_resolver.dart
import '../config/api_config.dart';

String resolveMediaUrl(String? value) {
  final url = value?.trim() ?? '';
  if (url.isEmpty) {
    return '';
  }

  if (url.startsWith('blob:')) {
    return '';
  }

  final parsed = Uri.tryParse(url);
  if (parsed?.scheme == 'http') {
    return parsed!.replace(scheme: 'https').toString();
  }

  if (parsed?.hasScheme == true) {
    return url;
  }

  final path = url.startsWith('/') ? url : '/$url';
  return '${ApiConfig.apiOrigin}$path';
}

String readMediaUrl(Map<String, dynamic> json, List<String> keys) {
  for (final key in keys) {
    final value = json[key]?.toString().trim() ?? '';
    if (value.isNotEmpty) {
      return resolveMediaUrl(value);
    }
  }

  return '';
}

File: src/apps/jhonny_home_studio_admin/src/app/core/utils/media-url-resolver.ts
import { ApiConfig } from '../config/api-config';

export function resolveMediaUrl(value?: string | null): string {
  const url = (value ?? '').trim();
  if (!url || url.startsWith('blob:')) {
    return '';
  }

  try {
    const parsed = new URL(url);
    if (parsed.protocol) {
      return url;
    }
  } catch {
    // Treat as relative below.
  }

  const normalizedPath = url.startsWith('/') ? url : `/${url}`;
  return `${ApiConfig.baseUrl}${normalizedPath}`;
}
